Jenkinsfile

```
@@ -852,58 +852,90 @@ def isChangedXPackCode(patterns) {
 }
 
 def loadConfigEnvVars(){
-  env.BUILD_AUDITBEAT = isChanged([
+  // Libbeat is the core framework of Beats. It has no additional dependencies
+  // on other projects in the Beats repository.
+  env.BUILD_LIBBEAT = isChangedOSSCode([])
+  env.BUILD_LIBBEAT_XPACK = env.BUILD_LIBBEAT || isChangedXPackCode([])
+
+  // Auditbeat depends on metricbeat as framework, but does not include any of
+  // the modules from Metricbeat.
+  // The Auditbeat x-pack build contains all functionality from OSS Auditbeat.
+  env.BUILD_AUDITBEAT = isChangedOSSCode([
     "^metricbeat/.*",
     "^auditbeat/.*",
   ])
-  env.BUILD_AUDITBEAT_XPACK = isChangedXPackCode([
-    "^metricbeat/.*",
-    "^auditbeat/.*",
+  env.BUILD_AUDITBEAT_XPACK = env.BUILD_AUDITBEAT || isChangedXPackCode([
     "^x-pack/auditbeat/.*",
   ])
+
+  // Dockerlogbeat is a standalone Beat that only relies on libbeat.
   env.BUILD_DOCKERLOGBEAT_XPACK = isChangedXPackCode([
     "^x-pack/dockerlogbeat/.*",
   ])
+
+  // Filebeat depends on libbeat only.
+  // The Filebeat x-pack build contains all functionality from OSS Filebeat.
   env.BUILD_FILEBEAT = isChangedOSSCode(["^filebeat/.*"])
-  env.BUILD_FILEBEAT_XPACK = isChangedXPackCode([
-    "^filebeat/.*",
+  env.BUILD_FILEBEAT_XPACK = env.BUILD_FILEBEAT || isChangedXPackCode([
     "^x-pack/filebeat/.*",
   ])
+
+  // Metricbeat depends on libbeat only.
+  // The Metricbeat x-pack build contains all functionality from OSS Metricbeat.
+  env.BUILD_METRICBEAT = isChangedOSSCode(["^metricbeat/.*"])
+  env.BUILD_METRICBEAT_XPACK = isChangedXPackCode([
```

Following the earlier auditbeat pattern, maybe it should be like
```suggestion
  env.BUILD_METRICBEAT_XPACK = env.BUILD_METRICBEAT || isChangedXPackCode([
```

---

right, I missed that one.